src/test/java/org/jenkinsci/test/acceptance/controller/WinstoneControllerTest.java

```diff
@@ -0,0 +1,46 @@
+package org.jenkinsci.test.acceptance.controller;
+
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.name.Names;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.File;
+import java.io.IOException;
+
+public class WinstoneControllerTest {
+    private Injector i;
+    private File tempFile;
+
+    @Before
+    public void setUp() {
+        i = Guice.createInjector(new AbstractModule() {
+            @Override
+            protected void configure() {
+                try {
+                    tempFile = File.createTempFile("junit", "test");
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                bind(String.class).annotatedWith(Names.named("WORKSPACE")).toInstance("/tmp/");
+                bind(String.class).annotatedWith(Names.named("quite")).toInstance("");
+                bind(File.class).annotatedWith(Names.named("form-element-path.hpi")).toInstance(tempFile);
+                bind(File.class).annotatedWith(Names.named("jenkins.war")).toInstance(tempFile);
+            }
+        });
+    }
+
+    @Test
+    public void constructWithJavaHome() {
+        File javaHome = new File("/tmp/path/to/java_home");
```

Comment:
please `/tmp` files and stuff on my machine when I run tests.   `/tmp` is NOT the temp path, and I do not want you to do that.  I happen to use windows so this makes `c:\tmp\path\to\java_home`, but even then the temp directory may not be the standard shared one than some (but not Nixes use).

the windows agents should actually throw an Exception here as you are trying to use something that is not appropriate - but I doubt we use them here, and I also doubt they are setup correctly to prevent this mishcheif 

Also who is cleaning up here